pico/types/src/main/java/io/helidon/pico/types/DefaultTypedElementName.java

```diff
@@ -0,0 +1,283 @@
+ */
+
+package io.helidon.pico.types;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Default implementation for {@link io.helidon.pico.types.TypedElementName}.
+ */
+public class DefaultTypedElementName implements TypedElementName {
+    private final TypeName typeName;
+    private final List<TypeName> componentTypeNames;
+    private final String elementName;
+    private final String defaultValue;
+    private final List<AnnotationAndValue> annotations;
+    private final List<AnnotationAndValue> elementTypeAnnotations;
+
+    protected DefaultTypedElementName(Builder b) {
+        this.typeName = b.typeName;
+        this.componentTypeNames = Objects.isNull(b.componentTypeNames)
+                ? Collections.emptyList() : Collections.unmodifiableList(b.componentTypeNames);
+        this.elementName = b.elementName;
+        this.defaultValue = b.defaultValue;
+        this.annotations = Objects.isNull(b.annotations)
+                ? Collections.emptyList() : Collections.unmodifiableList(b.annotations);
+        this.elementTypeAnnotations = Objects.isNull(b.elementTypeAnnotations)
+                ? Collections.emptyList() : Collections.unmodifiableList(b.elementTypeAnnotations);
+    }
+
+    @Override
+    public TypeName typeName() {
+        return typeName;
+    }
+
+    @Override
+    public String elementName() {
+        return elementName;
+    }
+
+    @Override
+    public Optional<String> defaultValue() {
+        return Optional.ofNullable(defaultValue);
+    }
+
+    @Override
+    public List<AnnotationAndValue> annotations() {
+        return annotations;
+    }
+
+    @Override
+    public List<AnnotationAndValue> elementTypeAnnotations() {
+        return elementTypeAnnotations;
+    }
+
+    /**
+     * Returns the component type names describing the element.
+     *
+     * @return the component type names of the element
+     */
+    public List<TypeName> getComponentTypeNames() {
+        return componentTypeNames;
+    }
+
+    @Override
+    public int hashCode() {
+        return System.identityHashCode(typeName());
+    }
+
+    @Override
+    public boolean equals(Object another) {
+        if (!(another instanceof TypedElementName)) {
+            return false;
+        }
+
+        TypedElementName other = (TypedElementName) another;
+        return Objects.equals(typeName(), other.typeName())
+                && Objects.equals(elementName(), other.elementName())
+                && Objects.equals(annotations(), other.annotations());
+    }
+
+    @Override
+    public String toString() {
+        return toDeclaration();
+    }
+
+    /**
+     * Provides a description for this instance.
+     *
+     * @return provides the {typeName}{space}{elementName}
+     */
+    public String toDeclaration() {
+//        if (Objects.nonNull(typeName()) && Objects.nonNull(elementName())) {
```

Comment:
Can we remove all this commented out code?

Reply:
removed and pushed.